client/components/admin/UserForm.jsx

```diff
@@ -0,0 +1,260 @@
+      formIsValid = false;
+    }
+    this.setState({ errors });
+    return formIsValid;
+  }
+
+  clearError(event) {
+    const field = event.target.name;
+    const errors = this.state.errors;
+
+    errors[field] = '';
+    const invalid = false;
+    this.setState({ errors, invalid });
+  }
+  render() {
+    const { errors } = this.state;
+    const { selectedUser } = this.props;
+    const hasValue = this.props.selectedUser;
+
+    const form = (
+      <div className="col s12 m12 z-depth-5 card-panel transparent80">
+        <form className="login-form">
+          <div className="row margin">
+            <TextInput
+              type="text"
+              name="name"
+              value={this.state.fullname}
+              placeholder="fullname"
+              icon="person_outline"
+              onChange={this.onChange}
+              error={errors.name}/>
+          </div>
+          <div className="row margin">
+            <TextInput
+              type="text"
+              name="username"
+              value={this.state.username}
+              placeholder="username"
+              icon="person"
+              onChange={this.onChange}
+              clearError={this.clearError}
+              error={errors.username}/>
+          </div>
+          <div className="row margin">
+            <TextInput
+              type="email"
+              name="email"
+              value={this.state.email}
+              placeholder="email"
+              icon="email"
+              onChange={this.onChange}
+              clearError={this.clearError}
+              error={errors.email}/>
+          </div>
+          {selectedUser ? '' :
+            <div className="row margin">
+              <TextInput
+                type="password"
+                name="password"
+                placeholder="password"
+                icon="lock"
+                onChange={this.onChange}
+                clearError={this.clearError}
+                error={errors.email}/>
+            </div>
+          }
+          <div className="row margin">
+          <label>User Role</label>
+          <div className="input-field col s12">
+            <select name="roleId" id="mySelectBox"
+            value={this.state.select}
+            className="browser-default" onChange={this.onChange}>
+            <option value="" disabled >User Role</option>
+            {this.props.allRoles.map(role =>
+               (<option key={role.id}
+              value={role.id}>{role.title}</option>)
+            )
+            }
+          </select>
+          </div>
+          </div>
+          <div className="row">
+            <div className="input-field">
+              <input
+                type="submit"
+                value="Save"
+                className="btn waves-effect waves-light teal darken-1"
+                onClick={hasValue ? this.updateUser : this.saveUser}/>
+            </div>
+          </div>
+        </form>
+      </div>
+    );
+    return (
```

Comment:
Is there a reason why you declared the form before returning it ? Why not return the  the whole element, instead of assigning it to a variable which is eventually returned as is.